lib/python/compliance_checker_test.py

```diff
@@ -0,0 +1,162 @@
+imos:1.3
+ghrsst:latest
+
+The script reads a json config file [CONFIG_FILE]. This config file lists for each NetCDF to test:
+ - The path of the NetCDF on the imos s3 bucket
+ - checks which should pass (checks_success)
+ - criteria to apply (normal, lenient ... default is normal)
+ - checks expected to fail (checks_fail)
+ - checks to skip (skip_checks" such as "check_convention_globals")
+
+Each file will be downloaded in a temporary folder. The compliance checker will be run on each of those NetCDF.
+
+The results will be output to a temporary directory (see outfile_path variable):
+* a json file with a similar structure to the CONFIG_FILE containing the different results
+* if some tests fail, the compliance checker output files will be saved in the same output directory
+
+If all test succeeded, the string 'False' should not exist in the output json file. If "False"
+is to be found, this means a test didn't output the required result.
+
+author: Besnard, Laurent
+"""
+import json
+import os
+import tempfile
+
+import cf_units
+import compliance_checker
+from six.moves.urllib.request import urlretrieve
+
+from util import pass_netcdf_checker
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+CONFIG_FILE = os.path.join(TEST_ROOT, "compliance_checker_imos_files_config.json")
+OUTPUT_DIR = tempfile.mkdtemp(prefix='compliance_checker_testing_results_')
+
+with open(CONFIG_FILE, 'r') as f:
+    compliance_config = json.load(f)
+
+print("compliance checker {cc_version}\ncf units {cf_version}".format(cc_version=compliance_checker.__version__,
+                                                                      cf_version=cf_units.__version__))
+
+
+def download_temporary_netcdf(url):
+    """
+    downloads NetCDF into a temporary folder
+    return path of NetCDF
+    """
+    tempfile_obj = tempfile.mkstemp()
+    tempfile_path = tempfile_obj[1]  # path of the downloaded NetCDF
+    urlretrieve(url, tempfile_path)
+
+    return tempfile_path
+
+
+def netcdf_tests_info(sub_collection):
+    """
+    parse sub collection dictionnary to retrieve essential information
+    return dictionnary
+    """
+    # handling default parameter for criteria
+    criteria = sub_collection[1]['check_params']['criteria']
+    if criteria == []:
+        criteria = "normal"
+    else:
+        criteria = criteria[0]
+
+    return {
+        'file_url': sub_collection[1]['path'][0],
+        'check_success_tests': sub_collection[1]['check_params']['checks_success'],
+        'check_fail_tests': sub_collection[1]['check_params']['checks_fail'],
+        'criteria': criteria,
+        'skip_checks': sub_collection[1]['check_params']['skip_checks']
+    }
+
+
+# collection is equivalent to a facility/sub-facility in the json file
+for collection in compliance_config:
+    print("Running test suite for: {collection}".format(collection=collection))
+
+    for sub_collection in compliance_config[collection].items():
+
+        def run_tests_netcdf(test_type):
```

Comment:
that's just a way to code. I used nested functions because I could share easily variables, without having to create a function with many arguments.

anyway, I refactored it to have independant functions